backend/core/src/csv/csv-builder.service.ts

```diff
@@ -1,130 +1,330 @@
+    const obj = {}
+    const fields = [
+      "firstName",
+      "middleName",
+      "lastName",
+      "type",
+      "agency",
+      "otherType",
+      "emailAddress",
+      "phoneNumber",
+    ]
+
+    fields.forEach((field, i) => {
+      obj[`alternateContact ${field}`] = `_${sort}${i}_Alternate Contact ${this.capAndSplit(field)}`
+    })
+
+    return obj
+  }
+
+  private capAndSplit(str: string): string {
+    let newStr = capitalizeFirstLetter(str)
+    newStr = newStr.split(/(?=[A-Z])/).join(" ")
+    return newStr
+  }
+
+  private setHeader(header: string, index: number): [string, string?] {
+    let newKey = header
+    // we pass along the mapped key for reference, so we can more easily set data to the correct column
+    const test1 = this.mappedFields[header]
+    const test2 = this.mappedFields[header.split(" ")[0]]
+    let mappedStr = test1 ?? test2
+    if (!test1 && test2) {
+      const [, ...fields] = header.split(" ")
+      if (fields) {
+        fields.forEach((field) => {
+          mappedStr += " " + this.capAndSplit(field)
+        })
       }
-      if (metadataObj.type === "array") {
-        const metadataArray: FormattingMetadataArray = metadataObj as FormattingMetadataArray
-        for (let i = 0; i < metadataArray.size; i++) {
-          let row
-          try {
-            row = value[i]
-          } catch (e) {
-            row = {}
-          }
-          const metadataArrayItems = metadataArray.items.map((item) =>
-            this.incrementMetadataArrayItemLabel(item, i + 1)
-          )
-          outputRows = outputRows.concat(this.flattenJson(row, metadataArrayItems))
-        }
-      } else {
-        const metadata: FormattingMetadata = metadataObj as FormattingMetadata
-        try {
-          outputRows.push(metadata.formatter(value))
-        } catch (e) {
-          outputRows.push("")
-        }
+    }
+
+    if (index !== 0) {
+      newKey += ` (${index})`
+      if (mappedStr) {
+        mappedStr += ` (${index})`
       }
     }
-    return outputRows
+    if (this.headers[newKey] === undefined) {
+      this.headers[newKey] = 1
+    }
+    return [newKey, mappedStr]
   }
 
-  private getHeaders(metadataArray: any[]) {
-    let headers: string[] = []
-    for (const metadata of metadataArray) {
-      if (metadata.type === "array") {
-        for (let i = 0; i < metadata.size; i++) {
-          const items = metadata.items.map((item) =>
-            this.incrementMetadataArrayItemLabel(item, i + 1)
-          )
-          headers = headers.concat(this.getHeaders(items))
+  private resetData() {
+    this.headers = {}
+    this.data = []
+  }
+
+  private setProgress(inProgress: boolean) {
+    this.inProgress = inProgress
+  }
+
+  private getExcludeKeysRegex() {
+    return new RegExp(this.excludeKeys.join("|"))
+  }
+
+  private addToExlucdedKeys(key) {
+    this.excludeKeys.push(key)
+  }
+
+  private parseApplication(key: string, val: any, header: string, index = 0): void {
```

Comment:
Method is called `parseApplication`, but the `val` is actually `any` so I think we should consider a better method name.

Reply:
How about parseData?